src/wh_client_crypto.c

```diff
@@ -2741,6 +2751,156 @@ int wh_Client_HkdfMakeExportKey(whClientContext* ctx, int hashType,
 
 #endif /* HAVE_HKDF */
 
+#ifdef HAVE_CMAC_KDF
+static int _CmacKdfMakeKey(whClientContext* ctx, whKeyId saltKeyId,
+                           const uint8_t* salt, uint32_t saltSz, whKeyId zKeyId,
+                           const uint8_t* z, uint32_t zSz,
+                           const uint8_t* fixedInfo, uint32_t fixedInfoSz,
+                           whNvmFlags flags, uint32_t label_len,
+                           const uint8_t* label, whKeyId* inout_key_id,
+                           uint8_t* out, uint32_t outSz)
+{
+    int                              ret     = WH_ERROR_OK;
+    uint8_t*                         dataPtr = NULL;
+    whMessageCrypto_CmacKdfRequest*  req     = NULL;
+    whMessageCrypto_CmacKdfResponse* res     = NULL;
+    uint16_t                         group   = WH_MESSAGE_GROUP_CRYPTO;
+    uint16_t                         action  = WC_ALGO_TYPE_KDF;
+    whKeyId                          key_id  = WH_KEYID_ERASED;
+
+    if ((ctx == NULL) || (outSz == 0)) {
+        return WH_ERROR_BADARGS;
+    }
+
+    if ((saltSz > 0 && salt == NULL) || (zSz > 0 && z == NULL) ||
+        (fixedInfoSz > 0 && fixedInfo == NULL)) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Retrieve the shared communication buffer */
+    dataPtr = wh_CommClient_GetDataPtr(ctx->comm);
+    if (dataPtr == NULL) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Prepare request structure with subtype information */
+    req = (whMessageCrypto_CmacKdfRequest*)_createCryptoRequestWithSubtype(
+        dataPtr, WC_ALGO_TYPE_KDF, WC_KDF_TYPE_TWOSTEP_CMAC);
+
+    uint32_t total_len = sizeof(whMessageCrypto_GenericRequestHeader) +
+                         sizeof(*req) + saltSz + zSz + fixedInfoSz;
+
+    if (total_len > WOLFHSM_CFG_COMM_DATA_LEN) {
+        return WH_ERROR_BADARGS;
+    }
+    uint16_t req_len = (uint16_t)total_len;
+
+    if (inout_key_id != NULL) {
+        key_id = *inout_key_id;
+    }
+
+    req->flags       = flags;
+    req->keyIdSalt   = saltKeyId;
+    req->keyIdZ      = zKeyId;
+    req->keyIdOut    = key_id;
+    req->saltSz      = saltSz;
+    req->zSz         = zSz;
+    req->fixedInfoSz = fixedInfoSz;
+    req->outSz       = outSz;
+
+    memset(req->label, 0, WH_NVM_LABEL_LEN);
+    if ((label != NULL) && (label_len > 0)) {
+        if (label_len > WH_NVM_LABEL_LEN) {
+            label_len = WH_NVM_LABEL_LEN;
+        }
+        memcpy(req->label, label, label_len);
+    }
+
+    uint8_t* payload = (uint8_t*)(req + 1);
+
+    if (saltSz > 0 && salt != NULL) {
+        memcpy(payload, salt, saltSz);
+        payload += saltSz;
+    }
+
+    if (zSz > 0 && z != NULL) {
+        memcpy(payload, z, zSz);
+        payload += zSz;
+    }
+
+    if (fixedInfoSz > 0 && fixedInfo != NULL) {
+        memcpy(payload, fixedInfo, fixedInfoSz);
+    }
```

Comment:
```suggestion
        memcpy(payload, fixedInfo, fixedInfoSz);
        payload += fixedInfoSz;
    }
    
    (void)payload;
```
Following the pattern.